oat-sa/tao-core
# Feature/AUT-3628/Apply the Solar Design to the buttons

### Related to: https://oat-sa.atlassian.net/browse/AUT-3646

### Summary

Apply the Solar Design to the buttons.

![image](https://github.com/oat-sa/tao-core/assets/1500098/5d59081a-8c31-48d0-a62f-e2e2ab381999)

### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

A small refactoring has been applied to the previously made implementation: split into more modules, small corrections, ...

Most of the buttons are now following the Solar Design. A few additions will come with the next ticket. 

### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3628/solar-design-buttons`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the buttons, compare with the UX mockups
